src/BinSkim.Sdk/CompilerDataLogger.cs

```diff
@@ -263,8 +228,7 @@ public void Write(BinaryAnalyzerContext context, CompilerData compilerData)
                 { "dialect", compilerData.Dialect },
                 { "moduleName", compilerData.ModuleName ?? string.Empty },
                 { "moduleLibrary", (compilerData.ModuleName == compilerData.ModuleLibrary ? string.Empty : compilerData.ModuleLibrary ?? string.Empty) },
-                { "sessionId", this.sessionId },
```

Comment:
>sessionId [](http://example.com/codeflow?start=19&length=9)

It's good to prefer AI's actually handling for session id, thanks.

I have wondered, however, if we should have previously had an additional id that would help us sort telemetry within the data store.

For example. think about telemetry that we send to the working system that strictly comprises test information: should we have a constant guid/id here that identifies that data comain?

Or is there a more natural AI mechanism to keep data sets isolated? Any suggestions? We have pre-release test data, produced by dev tests, nightly functionals. We will have data produced by release candidates (that will execute in production automation systems) that could be useful to identify. eg.

Reply:
Probably "role name" is closest, but I'm not sure I understand what you mean by "data domain".

Reply:
RIght, not super clear. What I mean is do we need a way of organizing data in the general AI telemetry channel in logical sets.  Perhaps by allocating a specific GUID that groups all telemetry that's ever collected which publishes the guid. So, we have one GUID that all of our testing/pre-release versions send with telemetry. We use another GUID for all official releases. All telemetry is persisted to the same AI sink using the same AI connection strings (because we can't easily reconfigure these when running pre-release vs. production, say in OneBranch or some other ADO pipeline).

Reply:
OK. RoleName is the closest to that. The AnalysisSummaryExtractor (called from CompilerDataLogger) already queries a few interesting environment variables such as the OrganizationName, ProjectName and RepositoryId. Ideal for when running in an AzDO pipeline. We could compose a RoleName from one or more of those environment variables.